packages/iov-bns/src/context.ts

```diff
@@ -83,21 +90,25 @@ export class Context {
   }
 
   public swapOfferFromTx(tx: ConfirmedTransaction<SwapCounterTransaction>): OpenSwap {
-    const counter: SwapCounterTransaction = tx.transaction;
+    const counterTransaction: SwapCounterTransaction = tx.transaction;
     // TODO: do we really want errors here, or just filter them out???
-    if (!isHashIdentifier(counter.hashCode)) {
+    if (!isHashIdentifier(counterTransaction.hashCode)) {
       throw new Error("swap not controlled by hash lock");
     }
+    const transactionCreator: PublicIdentity = {
```

Comment:
Nice! Pulling it from the Transaction...

Maybe there could be a helper if this is done many places?
`function transactionCreator(transaction: UnsignedTransaction): PublicIdentity`

Reply:
I have something different in mind that would clean up many many places: Convert `UnsignedTransaction.chainId` + `UnsignedTransaction.signer` into

```ts
interface UnsignedTransaction {
  creator: PublicIdentity;
  // ...
}
```

Creator is a better name when you think of multi signatures.

Reply:
Sure. More breaking to API, but that is fine now, since it is cleaner.

Not sure if it hides the fact the transaction is tied to one chain, but users should soon learn that PublicIdentities are chain-dependent.

Go for it, make this cleanup!

Reply:
I'm the biggest user... and deal with all breaking API changes in the wallet, so don't worry

Reply:
> Not sure if it hides the fact the transaction is tied to one chain

It is not unless it is signed. One could argue that an UnsignedTransaction should not be tied to a signer or a chain and only get this connection at signing time. But we create those _to be signed by one identity_ and thus fix signer/chain before the signature comes in, which is also good.

Reply:
https://github.com/iov-one/iov-core/issues/640